File: src/domain/enums.py
from __future__ import annotations

from enum import StrEnum


class RunStatus(StrEnum):
    QUEUED = "queued"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class RunType(StrEnum):
    PROMPT_EVAL = "prompt_eval"
    RAG_EVAL = "rag_eval"
    COMPARISON_BACKING_RUN = "comparison_backing_run"
    SUGGESTION_EVAL = "suggestion_eval"


class TriggerSource(StrEnum):
    MCP = "mcp"
    API = "api"
    CLI = "cli"
    SCHEDULE = "schedule"
    CI = "ci"


class MetricDirection(StrEnum):
    HIGHER_IS_BETTER = "higher_is_better"
    LOWER_IS_BETTER = "lower_is_better"


class ArtifactType(StrEnum):
    TRANSCRIPT = "transcript"
    RETRIEVED_CONTEXT = "retrieved_context"
    PROMPTFOO_REPORT = "promptfoo_report"
    COMPARISON_REPORT = "comparison_report"
    DEBUG_BUNDLE = "debug_bundle"


class CaseResultStatus(StrEnum):
    PASSED = "passed"
    FAILED = "failed"
    ERROR = "error"
    SKIPPED = "skipped"

File: pkg/domain/enums.go
package domain

type RunStatus string

const (
	StatusQueued    RunStatus = "queued"
	StatusRunning   RunStatus = "running"
	StatusCompleted RunStatus = "completed"
	StatusFailed    RunStatus = "failed"
	StatusCancelled RunStatus = "cancelled"
)

type RunType string

const (
	RunTypePromptEval            RunType = "prompt_eval"
	RunTypeRagEval               RunType = "rag_eval"
	RunTypeComparisonBackingRun RunType = "comparison_backing_run"
	RunTypeSuggestionEval        RunType = "suggestion_eval"
)

type TriggerSource string

const (
	TriggerSourceMCP      TriggerSource = "mcp"
	TriggerSourceAPI      TriggerSource = "api"
	TriggerSourceCLI      TriggerSource = "cli"
	TriggerSourceSchedule TriggerSource = "schedule"
	TriggerSourceCI       TriggerSource = "ci"
)

type MetricDirection string

const (
	HigherIsBetter MetricDirection = "higher_is_better"
	LowerIsBetter  MetricDirection = "lower_is_better"
)

type ArtifactType string

const (
	ArtifactTranscript       ArtifactType = "transcript"
	ArtifactRetrievedContext ArtifactType = "retrieved_context"
	ArtifactPromptfooReport  ArtifactType = "promptfoo_report"
	ArtifactComparisonReport ArtifactType = "comparison_report"
	ArtifactDebugBundle      ArtifactType = "debug_bundle"
)

type CaseResultStatus string

const (
	CasePassed  CaseResultStatus = "passed"
	CaseFailed  CaseResultStatus = "failed"
	CaseError   CaseResultStatus = "error"
	CaseSkipped CaseResultStatus = "skipped"
)
